src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -209,6 +242,12 @@ string EncodeUrl (Uri url)
 			while (true) {
 				URL java_url = new URL (EncodeUrl (redirectState.NewUrl));
 				URLConnection java_connection = java_url.OpenConnection ();
+				if (ConnectTimeout != TimeSpan.Zero)
+					java_connection.ConnectTimeout = (int)ConnectTimeout.TotalMilliseconds;
```

Comment:
This should probably be `checked ((int) ConnectTimeout.TotalMilliseconds)`, in case the value overflows.